# In this Jupyter Notebook I will read in the data that was given and calculate what a qualifying offer would be, based on the average of the top 125 salaries in the MLB.

In [1]:
import pandas as pd

## First, I read in the data that I converted into a CSV file. In excel where I converted the file I made all the values integers and "no salary data" 0. 

In [2]:
df = pd.read_csv("data_phillies.csv")
df

Player      Salary  Year Level
0            Abreu, Jose  11666667.0  2016   MLB
1        Acevedo, Andury    507500.0  2016   MLB
2         Ackley, Dustin   3200000.0  2016   MLB
3      Adames, Cristhian    507500.0  2016   MLB
4          Adams, Austin    507500.0  2016   MLB
...                  ...         ...   ...   ...
1203     Zimmerman, Ryan  14000000.0  2016   MLB
1204  Zimmermann, Jordan  18000000.0  2016   MLB
1205        Zobrist, Ben  10500000.0  2016   MLB
1206        Zunino, Mike    507500.0  2016   MLB
1207          Zych, Tony    507500.0  2016   MLB

[1208 rows x 4 columns]

## Next, I select the salary column because that is what we are interested in analyzing.

In [3]:
just_salary = df[["Salary"]]
just_salary 

Salary
0     11666667.0
1       507500.0
2      3200000.0
3       507500.0
4       507500.0
...          ...
1203  14000000.0
1204  18000000.0
1205  10500000.0
1206    507500.0
1207    507500.0

[1208 rows x 1 columns]

## Then, I dropped any null values from the column to only leave us with the numbers we are intersted in.

In [4]:
masked = just_salary[just_salary["Salary"].notna()]
masked.dropna(how = "any")
masked

Salary
0     11666667.0
1       507500.0
2      3200000.0
3       507500.0
4       507500.0
...          ...
1203  14000000.0
1204  18000000.0
1205  10500000.0
1206    507500.0
1207    507500.0

[1196 rows x 1 columns]

## I then put the Data Frame into descending order, which means the highest value is at the top, counting down. I took the highest 125 values and I put that into a new Data Frame. 

In [6]:
descending = just_salary.sort_values(by="Salary", ascending = False)
top_125 = descending.head(125)
top_125

Salary
583   34571429.0
449   34000000.0
871   30000000.0
471   28410000.0
1133  28000000.0
...          ...
476   10000000.0
517   10000000.0
257   10000000.0
580   10000000.0
146    9950000.0

[125 rows x 1 columns]

## Finally, I calculate the mean of these 125 values through Pandas built in mean function.

In [7]:
qual_offer = (top_125.astype('float32').mean(axis = 0))
qual_offer

Salary    16518994.0
dtype: float32

From this, we can see the qualifying offer for 2016 that we calculated is: 16,518,994. From a quick google search, I found that the actual qualifying offer in 2016 was 17.2 million, which is close to what we got here. If it wasn't for the some players having no salary data or null values for their salary, the figure should line up exactly. 